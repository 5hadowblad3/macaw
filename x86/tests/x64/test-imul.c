#include "util.h"

int g = -11;

void _start() {
  asm("mov %edx,0xc(%rsp)\n"
      "lea (%r12,%rax,1),%ecx\n"
      "xor %eax,%eax\n"
      "imul $0xffffffe8,%ebx,%ebx\n"
      "add %edi,%ebx\n"
      "mov %edx,%edi\n"
    );

  EXIT();
}
