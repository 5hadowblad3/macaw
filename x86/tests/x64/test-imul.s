	.file	"test-imul.c"
	.text
	.globl	g
	.data
	.align 4
	.type	g, @object
	.size	g, 4
g:
	.long	-11
	.text
	.globl	_start
	.type	_start, @function
_start:
.LFB0:
	.cfi_startproc
	pushq	%rbp
	.cfi_def_cfa_offset 16
	.cfi_offset 6, -16
	movq	%rsp, %rbp
	.cfi_def_cfa_register 6
#APP
# 6 "test-imul.c" 1
	mov %edx,0xc(%rsp)
lea (%r12,%rax,1),%ecx
xor %eax,%eax
imul $0xffffffe8,%ebx,%ebx
add %edi,%ebx
mov %edx,%edi

# 0 "" 2
# 14 "test-imul.c" 1
	movq $60,%rax
movq $0,%rdi
syscall
# 0 "" 2
#NO_APP
	nop
	popq	%rbp
	.cfi_def_cfa 7, 8
	ret
	.cfi_endproc
.LFE0:
	.size	_start, .-_start
	.ident	"GCC: (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0"
	.section	.note.GNU-stack,"",@progbits
