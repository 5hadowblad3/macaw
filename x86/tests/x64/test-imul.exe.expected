R { funcs = [(Addr 1 0x2b1, [(Addr 1 0x2b1, 37)])]
  , ignoreBlocks = [Addr 1 0x2d6]
  }
